versioned/storage/common/src/main/java/org/projectnessie/versioned/storage/common/logic/CommitLogicImpl.java

```diff
@@ -1037,11 +1036,17 @@ public CreateCommit.Builder diffToCreateCommit(
         // key removed
         createCommit.addRemoves(
             commitRemove(d.key(), d.fromPayload(), requireNonNull(d.fromId()), d.fromContentId()));
-      } else {
+      } else if (Objects.equals(d.fromContentId(), d.toContentId())) {
         // key updated
         createCommit.addAdds(
             commitAdd(
-                d.key(), d.toPayload(), requireNonNull(d.toId()), d.fromId(), d.fromContentId()));
+                d.key(), d.toPayload(), requireNonNull(d.toId()), d.fromId(), d.toContentId()));
+      } else {
+        // key re-added
+        createCommit.addRemoves(
```

Comment:
Does this not break proper merge errors in case the same key was separately added on source and destination branches?

Reply:
:+1: